nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/QueryElasticsearchHttp.java

```diff
@@ -0,0 +1,410 @@
+public class QueryElasticsearchHttp extends AbstractElasticsearchHttpProcessor {
+
+    private static final String FIELD_INCLUDE_QUERY_PARAM = "_source_include";
+    private static final String QUERY_QUERY_PARAM = "q";
+    private static final String SORT_QUERY_PARAM = "sort";
+    private static final String FROM_QUERY_PARAM = "from";
+    private static final String SIZE_QUERY_PARAM = "size";
+
+    public static final String TARGET_FLOW_FILE_CONTENT = "Flow file content";
+    public static final String TARGET_FLOW_FILE_ATTRIBUTES = "Flow file attributes";
+    private static final String ATTRIBUTE_PREFIX = "es.result.";
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description(
+                    "All FlowFiles that are read from Elasticsearch are routed to this relationship.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description(
+                    "All FlowFiles that cannot be read from Elasticsearch are routed to this relationship. Note that only incoming "
+                            + "flow files will be routed to failure.").build();
+
+    public static final Relationship REL_RETRY = new Relationship.Builder()
+            .name("retry")
+            .description(
+                    "A FlowFile is routed to this relationship if the document cannot be fetched but attempting the operation again may "
+                            + "succeed. Note that if the processor has no incoming connections, flow files may still be sent to this relationship "
+                            + "based on the processor properties and the results of the fetch operation.")
+            .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("query-es-query").displayName("Query")
+            .description("The Lucene-style query to run against ElasticSearch").required(true)
+            .expressionLanguageSupported(true).addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("query-es-index").displayName("Index")
+            .description("The name of the index to read from").required(true)
+            .expressionLanguageSupported(true).addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("query-es-type")
+            .displayName("Type")
+            .description(
+                    "The (optional) type of this document, used by Elasticsearch for indexing and searching. If the property is empty or set "
+                            + "to _all, the first document matching the identifier across all types will be retrieved.")
```

Comment:
You're right.. I copied this from FetchElasticsearchHttp, and the bug exists there as well.  It looks like the original author had the logic backwards between index and type: https://www.elastic.co/guide/en/elasticsearch/guide/current/multi-index-multi-type.html

I think we should change the "Index" property to use the _all syntax if empty or if "_all" is specified.  It looks like the ES syntax for searching all types is to simply leave out the "Type" path parameter.  I'll address this in all 3 processors when I get a chance.